scripts/scil_split_volume_labels.py

```diff
@@ -0,0 +1,178 @@
+
+or
+
+scil_split_volume_labels.py img.nii.gz labels -h
+
+
+IMPORTANT: your label image must be of an integer type.
+"""
+
+IDS_DESCRIPTION = """
+Split a label image into multiple images where the name of the output images
+is the id of the label (ex. 35.nii.gz, 36.nii.gz, ...). If the --range option
+is not provided, all labels of the image are extracted.
+"""
+
+LABELS_DESCRIPTION = """
+Split a label image into multiple images where the name of the output images
+is taken from a lookup table (ex: left-lateral-occipital.nii.gz,
+right-thalamus.nii.gz, ...). Only the labels included in the lookup table
+are extracted.
+
+New lookup tables can be added by adding new files to the
+scilpy/data/LUT/ directory.
+"""
+
+
+# Taken from http://stackoverflow.com/a/6512463
+def parseNumList(str_to_parse):
+    m = re.match(r'(\d+)(?:-(\d+))?$', str_to_parse)
+
+    if not m:
+        raise argparse.ArgumentTypeError("'" + str_to_parse + "' is not a " +
+                                         "range of numbers. Expected forms " +
+                                         "like '0-5' or '2'.")
+
+    start = m.group(1)
+    end = m.group(2) or start
+
+    start = int(start, 10)
+    end = int(end, 10)
+
+    if end < start:
+        raise argparse.ArgumentTypeError("Range elements incorrectly " +
+                                         "ordered in '" + str_to_parse + "'.")
+
+    return list(range(start, end+1))
+
+
+def _build_args_parser(luts):
+
+    p = argparse.ArgumentParser(
+        description=DESCRIPTION,
+        formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('label_image',
+                   help='Path of the input label file, '
+                        'in a format supported by Nibabel.')
+    p.add_argument('--output_dir', default='',
+                   help='Put all ouptput images in a specific directory.')
+    p.add_argument('--output_prefix', default='',
+                   help='Prefix to be used for each output image.')
+
+    mode_subparsers = p.add_subparsers(dest='mode',
+                                       help='Output naming schemes.')
+
+    ids_subparser = mode_subparsers.add_parser(
+        'ids', description=IDS_DESCRIPTION,
+        help='Names files using their label id.')
+    ids_subparser.add_argument('-r', '--range', type=parseNumList, nargs='*',
+                               help='Specifies a subset of labels to split, '
+                                    'formatted as 1-3 or 3 4.')
+
+    labels_subparser = mode_subparsers.add_parser(
+        'labels', help='Names the files using a lookup table.')
+    labels_subparser.add_argument(
+        'lut', choices=luts,
+        help='Lookup table, in the file ../data/LUT, '
+             'used to name the output files.')
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    # Get the valid LUT choices.
+    path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        '../data/LUT')
```

Comment:
This path doesn't work for me and make everything crash.

Reply:
Ok this is much more complex than I thought. Before everyone was git clone the repo.
Now we will have pip install scilpy (soon), setup develop, pip install -e ./ (or something).

These different way to install it give rise to many case. I personally dont know how to make it clean.

I suggest the input argument json, that we will still put in the repo, and the user will have to link it. This is likely safer.

Reply:
What is the error? I don't see why we can't make this work with a relative path?

Reply:
I would suggest using pathlib to make this easier (python 3.4+):
`pathlib.Path(__file__).parent / "../data/LUT"`

Reply:
Ok, just to make this clear for everyone, here is the problem:
Using relative paths in a script doesn't work, because the scripts may be installed in a separate directory as binaries (so it's impossible to find the module's location from the location of the script).
Thankfully, we can use the `inspect` module to find the location of the `scilpy` module:

`import inspect
import pathlib
import scilpy

module_path = inspect.getfile(scilpy)  # /home/USERNAME/git/scilpy/scilpy/__init__.py
lut_dir = str(pathlib.Path(module_path).parent.parent / "data/LUT")
`

Reply:
Thanks @ppoulin91 that was indeed the original reason for using the hardcoded path.